parlai/core/teachers.py

```diff
@@ -1944,6 +1904,10 @@ def shutdown(self):
         for t in self.tasks:
             t.shutdown()
 
+    def update_counters(self):
```

Comment:
Lots of new code here to update counters, and it seems to ultimately be syncing metrics. I suppose this is needed because the whole idea of this working is predicated on local metrics always being up-to-date?